test/extended/image_ecosystem/s2i_perl.go

```diff
@@ -58,7 +58,10 @@ var _ = g.Describe("[image_ecosystem][perl][Slow] hot deploy for openshift perl
 				_, err := exutil.WaitForPods(oc.KubeClient().Core().Pods(oc.Namespace()), dcLabel, exutil.CheckPodIsRunningFn, 1, 4*time.Minute)
 				o.ExpectWithOffset(1, err).NotTo(o.HaveOccurred())
 
-				result, err := CheckPageContains(oc, dcName, "", pageCountFn(i))
+				result, err := CheckPageContains(oc, dcName, "", pageCountFn(i), i)
+				if err != nil || !result {
+					exutil.DumpApplicationPodLogs(dcName, oc)
```

Comment:
fyi i've added log dumping for all pods in the namespace on any test failure, to all our extended tests, so i think you can remove these.